juju/juju
# Intern packaging package

The packaging package hasn't been maintained and the abstraction isn't holding its weight. In light of trying to own our dependencies and understand what we're installing at what stage, we should bring this in as some global constants don't make sense.

~~For instance, the UbuntuCloudArchiveSigningKey isn't used anywhere in the code base and we should probably remove it. Zapper is not referenced from Juju so could also be removed.~~ Removed.

Additionally, do we need the list of cloud archive packages set directly in the code? The code feels like there were grand plans for this, but was never fully implemented. As we move forward we could start removing parts of the abstraction that don't make sense.




## Checklist




## QA steps

Ensure that we can still bootstrap and deploy a charm.

```sh
$ juju bootstrap lxd test
$ juju add-model default
$ juju deploy ubuntu
```

## Links


**Jira card:** [JUJU-6071](https://warthogs.atlassian.net/browse/JUJU-6071)



[JUJU-6071]: https://warthogs.atlassian.net/browse/JUJU-6071?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
